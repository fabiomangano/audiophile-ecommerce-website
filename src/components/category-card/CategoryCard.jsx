import React from 'react';
import PropTypes from 'prop-types';
import style from './CategoryCard.module.scss';
import Button from '../button/Button';

const CategoryCard = ({ category, url, onClick }) => {
  return (
    <div className={style.container}>
      <img className={style.image} src={url} alt={category} />
      <h6>{category}</h6>
      <Button variant="link" onClick={onClick}>
        SHOP {'>'}
      </Button>
    </div>
  );
};

CategoryCard.propTypes = {
  category: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default CategoryCard;
