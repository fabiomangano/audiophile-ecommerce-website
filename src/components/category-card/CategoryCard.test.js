import React from 'react';
import { create } from 'react-test-renderer';
import CategoryCard from './CategoryCard';

describe('CategoryCard', () => {
  it('should correctly render', () => {
    const tree = create(
      <CategoryCard
        category="some-category"
        url="some-url"
        onClick={jest.fn()}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
