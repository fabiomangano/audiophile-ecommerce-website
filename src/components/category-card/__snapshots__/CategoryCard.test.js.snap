// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard should correctly render 1`] = `
<div
  className="container"
>
  <img
    alt="some-category"
    className="image"
    src="some-url"
  />
  <h6>
    some-category
  </h6>
  <button
    className="link md"
    onClick={[MockFunction]}
    size="md"
    type="button"
    variant="link"
  >
    SHOP 
    &gt;
  </button>
</div>
`;
